File: apps/macos/Sources/SecureStoreAdapter.swift
import Foundation

final class SecureStoreAdapter: SecureStore {
    private var map: [String: String] = [:]

    func get(_ key: String) -> String? {
        return map[key]
    }

    func set(_ key: String, value: String) {
        map[key] = value
    }

    func delete(_ key: String) {
        map.removeValue(forKey: key)
    }
}

File: apps/android/src/main/java/com/clipboardsync/SecureStoreAdapter.kt
package com.clipboardsync

class SecureStoreAdapter : SecureStore {
    private val map = mutableMapOf<String, String>()

    override fun get(key: String): String? = map[key]

    override fun set(key: String, value: String) {
        map[key] = value
    }

    override fun delete(key: String) {
        map.remove(key)
    }
}
